boards/Kconfig.v2

```diff
@@ -15,4 +16,13 @@ config BOARD_$(BOARD_FULL_STRING)
 	help
 	  Kconfig symbol identifying the board including full board identifier.
 
+config BOARD_QUALIFIERS
```

Comment:
technically, the `BOARD_QUALIFIERS` is not required to be in this file, it could go directly to `<boards>/Kconfig`, and would be empty for a v1 board.

One reason the `BOARD_$(BOARD_STRING)` and `BOARD_$(BOARD_FULL_STRING)` must be in Kconfig.v2 is because those may collide with `config BOARD_<name>` names used for v1 boards, as those used to be defined in `<board-dir>/Kconfig.board` and exists inside the choice.

For example like in:
https://github.com/zephyrproject-rtos/zephyr/blob/468eb56cf242eedba62006ee758700ee6148763f/boards/arm/nrf9160dk_nrf9160/Kconfig.board#L8
https://github.com/zephyrproject-rtos/zephyr/blob/468eb56cf242eedba62006ee758700ee6148763f/boards/arm/nrf9160dk_nrf9160/Kconfig.board#L11

But the qualifier can be moved to `<boards>/Kconfig` later when support for v1 is removed.